src/main/kotlin/ru/buhinder/alcopartyservice/repository/EventDaoFacade.kt

```diff
@@ -44,8 +44,29 @@ class EventDaoFacade(
         return eventRepository.findByIdAndNotPrivateAndAlcoholicIsNotBanned(eventId, alcoholicId)
     }
 
-    fun findByIdAndAlcoholicIsNotBanned(eventId: UUID, alcoholicId: UUID): Mono<EventEntity> {
-        return eventRepository.findByIdAndAlcoholicIsNotBanned(eventId, alcoholicId)
+    fun getByIdAndAlcoholicIsNotBannedAndStatusNotEnded(eventId: UUID, alcoholicId: UUID): Mono<EventEntity> {
+        return eventRepository.findByIdAndAlcoholicIsNotBannedAndStatusNotEnded(eventId, alcoholicId)
+            .switchIfEmpty {
+                Mono.error(
+                    EntityNotFoundException(
+                        message = "Event not found",
+                        payload = mapOf("id" to eventId)
+                    )
+                )
+            }
+    }
+
+    /// TODO: 05/08/2022 move to invitation link dao facade
+    fun getByInvitationLinkAndNotEnded(invitationLink: UUID): Mono<EventEntity> {
+        return eventRepository.findByInvitationLinkAndNotEnded(invitationLink)
+            .switchIfEmpty {
+                Mono.error(
+                    EntityNotFoundException(
+                        message = "Event not found",
```

Comment:
the message should be Invitation link not found, get method should be moved to invitation link dao facade